src/include/ikarus/Grids/GridEntities/DefaultGridEntities.h

```diff
@@ -286,46 +235,30 @@ namespace Ikarus::Grid {
     }
 
     const auto& getChildVertices() const { return std::get<0>(entitiesChildren); }
-
     auto& getChildVertices() { return std::get<0>(entitiesChildren); }
 
     /** \brief Returns the number of subEntities of this entity, e.g. a line has two verteces as
      * subtypes */
-    [[nodiscard]] unsigned int subEntities(unsigned int codim) const {
-      assert(codim > 0 && codim <= 2 && "Two dimensional entities only have subentities in 0<codimension<=2.");
-
-      if constexpr (mydimension == 1) return getChildVertices().size();
-      if constexpr (mydimension == 2) {
-        if (codim == 1)
-          return getChildEntities<1>().size();
-        else if (codim == 2)
-          return getChildEntities<0>().size();
-      }
-    }
+    [[nodiscard]] unsigned int subEntities(unsigned int codim) const;
+
     /** \brief Return the fundamental geometric type of the entity */
     [[nodiscard]] Dune::GeometryType type() const;
 
     /** \brief Returns the geometric realization of the entity */
-    auto geometry() const {
-      std::vector<Dune::FieldVector<double, dimensionworld>> fieldVectorVector;
-      for (const auto& pos :
-           std::ranges::transform_view(getChildVertices(), &DefaultGridEntity<griddim, griddim, wdim>::getPosition))
-        fieldVectorVector.push_back(toFieldVector(pos));
+    auto geometry() const;
 
-      return Ikarus::Geometry::IGeometry(Geometry(type(), fieldVectorVector));
-    }
     /** \brief Return copy of the id of this entity */
   private:
+    /** \brief The refinement level to which this entity belongs */
+    int levelIndex{};
+
     /** \brief A persistent id of this entity*/
     size_t id{};
 
     /** \brief Childrens of the entity on the current grid , i.e. surfaces of a cube*/
     decltype(Impl::ChildEntityPointerTupleGenerator<dimension, mydimension, dimensionworld>(
         std::make_integer_sequence<int, mydimension>())) entitiesChildren;
 
-    /** \brief Childrens of the entity on the current grid , i.e. surfaces of a cube*/
-    EntitiesFatherType entitiesFathers;
-
     /** \brief Childrens of the entity on a finer grid , i.e. subcubes of a cube*/
     std::vector<DefaultGridEntity<dimension, codimension, dimensionworld>*> levelEntitiesChildren;

```

Comment:
This is only here for potential later use for multigrid methods. Should i remove this?

Reply:
I'm talking here about Line 263.